CASMINST-55245 - change references to csm versions again in release 1… (#2889)
Cray-HPE/docs-csm: upgrade/1.2/README.md

from top to bottom. The content on this top-level page is meant to be terse. For additional reference material on the upgrade processes and scripts
mentioned explicitly on this page, see [resource material](resource_material/README.md).

> If you are considering an upgrade from CSM 1.0.x to 1.2.0, please instead download and use the latest released version of CSM 1.2.x.

Reviewer: ```suggestion
If upgrading from CSM 1.0.x to 1.2.0, then instead upgrade directly to the latest released version of CSM 1.2.x.
```
